Sources/SQLKitExtras/SQLKitExtras/Expressions/SQLCastExpression.swift

```diff
@@ -0,0 +1,63 @@
+import SQLKit
+
+/// Represents a `CAST` expression in SQL, which converts a value from one type to another.
+public struct SQLCastExpression: SQLExpression {
+    /// The original expression to be cast.
+    public let original: any SQLExpression
+
+    /// The desired type to cast the original expression to.
+    public let desiredType: any SQLExpression
+
+    /// Create a new ``SQLCast``.
+    ///
+    /// - Parameters:
+    ///   - original: The original expression to be cast.
+    ///   - desiredType: The desired type to cast the original expression to.
+    public init(expr: some SQLExpression, castType: any SQLExpression) {
+        self.original = expr
+        self.desiredType = castType
+    }
+
+    /// Convenience initializer for creating a ``SQLCast`` with a string for the original expression and a string for the desired type.
+    ///
+    /// - Parameters:
+    ///   - original: The original expression to be cast.
+    ///   - desiredType: The desired type to cast the original expression to, represented as a string.
+    public init(_ original: some SQLExpression, to type: some StringProtocol) {
+        self.init(expr: original, castType: .identifier(type))
+    }
+
+    /// See `SQLExpression.serialize(to:)`.
+    public func serialize(to serializer: inout SQLSerializer) {
+        serializer.statement {
+            $0.append("CAST(", self.original)
+            $0.append("AS", self.desiredType, ")")
+        }
```

Comment:
This introduces undesirable whitespace into the resulting SQL. Use `SQLFunction` and misappropriate `SQLAlias` (which already has the `AS` syntax) instead:
```suggestion
        SQLFunction("CAST", args: SQLAlias(self.original, as: self.desiredType)).serialize(to: &serializer)
```
(and adjust the tests accordingly, of course)